src/Components/Activities/index.jsx

```diff
@@ -9,39 +9,26 @@ import Modal from '../Shared/Modal';
 import Loader from '../Shared/Loader';
 
 function Activities() {
-  const { data, isPending, error } = useSelector((state) => state.activities);
-  const [message, setMessage] = useState('');
-  const [deleteModal, setDeleteModal] = useState(false);
-  const [confirmModal, setConfirmModal] = useState(false);
+  const { isPending, message, success, error } = useSelector((state) => state.activities);
+  const [modalMessage, setModalMessage] = useState('');
   const [showModal, setShowModal] = useState(false);
-  const location = useLocation();
-  const history = useHistory();
   const dispatch = useDispatch();
 
+  const handleModal = () => {
+    setShowModal(!showModal);
+    setTimeout(() => {
+      setShowModal();
+    }, 2000);
+  };
+
   useEffect(() => {
-    if (location.state) {
-      setShowModal(!showModal);
-      setMessage(location.state.message);
+    if (success) {
+      handleModal();
+      setModalMessage(message);
     }
-    history.replace({ ...history.location, state: undefined });
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const onDelete = async (id) => {
-    try {
-      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/${id}`, {
-        method: 'DELETE'
-      });
-      const data = await res.json();
-      setMessage(data.message);
-      setConfirmModal(!confirmModal);
-      setDeleteModal(!deleteModal);
-      return data;
-    } catch (error) {
-      console.error(error);
-    }
-  };
-
   useEffect(() => {
     getActivities(dispatch);
   }, [dispatch]);
```

Comment:
dispatch is a constant (function), it doesn't make sense to be listed as a dependency because it doesn't change state

Reply:
You're right, it doesn't change anything, I put it like that because the tutor was using it like that in the class